### PR TITLE
Update precompiled header to fix compatibility with latest Cocoapods

### PR DESCRIPTION
**Pull request by bradserbuddy**
Clean up .podspec.

### CODE REVIEW

**Comment by danaucpe** [comment]
nit: adding a .h import, but not using it anywhere?  or is that part of changing to the BaseObjects?  Other than that it looks fine to me.

**Comment by bradserbuddy** [comment]
tldr: it's needed by Cocoapods. BaseObjects is old code that's been deleted.

When a user adds the BuddySDK to a project via Cocoapods, Cocoapods brings down the SDK source from Github and builds it locally, and adds the library to the user's project. For some reason on the latest (few) builds of Cocoapods, that particular private header wasn't found by Cocoapods fairly obtuse local build process for building Swift-compatible libraries (specifically the Cocoapods-generated umbrella header couldn't find it; for more details see http://blog.cocoapods.org/Pod-Authors-Guide-to-CocoaPods-Frameworks/); adding it to the precompiled header allowed it to be found. It's not clear why that's the case, since sibling headers can be found, and the directory is listed in the Cocopods project. The other option was to make that header public, but, then it would show up in the wrong place in the auto-generated Cocoapods documentation. Good times.

**Comment by bradserbuddy** [comment]
The adding-it-do-the-precomplied header workaround\fix was mentioned in the Cocoapods mailing list.
